packages/demo-protobuf/src/registry.ts

```diff
@@ -0,0 +1,21 @@
+import protobuf from "protobufjs";
+
+import { cosmos_sdk as cosmosSdk, google } from "./generated/codecimpl";
+
+export class Registry {
+  private readonly types: Map<string, protobuf.Type>;
+
+  constructor(customTypes: readonly [string, protobuf.Type][] = []) {
+    this.types = new Map<string, protobuf.Type>([
+      ["/cosmos.Coin", (cosmosSdk.v1.Coin as unknown) as protobuf.Type],
```

Comment:
This does not seem to be a safe cast, as `protobuf.Type` contains all the dynamic stuff like

```ts
    /** Message fields. */
    public fields: { [k: string]: Field };

    /** Oneofs declared within this namespace, if any. */
    public oneofs: { [k: string]: OneOf };

    /** Extension ranges, if any. */
    public extensions: number[][];

    /** Reserved ranges, if any. */
    public reserved: (number[]|string)[];

    /** Message fields by id. */
    public readonly fieldsById: { [k: number]: Field };

    /** Fields of this message as an array for iteration. */
    public readonly fieldsArray: Field[];

    /** Oneofs of this message as an array for iteration. */
    public readonly oneofsArray: OneOf[];
```

which does not exist in the generated classes